### PR TITLE
Self versioning updater

### LINKED ISSUE #40: Provide own versioning updater
https://github.com/kachick/renovate-config-asdf/issues/36
https://github.com/renovatebot/renovate/issues/13308

### PR DESCRIPTION
**Pull request by kachick**
Fixes #40

However looks conflicted with #208 😮

### CODE REVIEW

**Comment by maennchen** [comment]
@kachick Oops

No worries, I'll just close mine since yours is more complete with tests etc.

File: .tool-versions [hunk 1]
```diff
@@ -1,3 +1,3 @@
-nodejs 18.11.0
+nodejs 18.10.0
```
Comment (kachick) [inline .tool-versions:1]: We should wait node-build merging for latest 🤔 

https://github.com/nodenv/node-build/pull/756

**Comment by kachick** [comment]
```
WARN: Config Warning
{
  "configError": {
    "validationError": "Cannot find preset's package (github>kachick/renovate-config-asdf:self)",
    "message": "config-validation",
    "stack": "Error: config-validation\n    at fetchPreset (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/presets/index.js:293:23)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Object.resolveConfigPresets (/home/ubuntu/renovateapp/node_modules/renovate/dist/config/presets/index.js:238:39)\n    at async mergeRenovateConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/merge.js:206:61)\n    at async getRepoConfig (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/config.js:15:14)\n    at async initRepo (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/init/index.js:33:14)\n    at async Object.renovateRepository (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/index.js:40:18)\n    at async renovateRepository (/home/ubuntu/renovateapp/app/worker/index.js:264:26)\n    at async /home/ubuntu/renovateapp/app/worker/index.js:458:5"
  },
  "res": "created"
}
```

🤔 

https://github.com/kachick/deploy-yard-to-pages/issues/13

**Comment by kachick** [comment]
https://github.com/kachick/renovate-config-asdf/commit/f8f4236e5d57eb5f37da08057f52b673f057ebbf fixed 🎉  https://github.com/kachick/deploy-yard-to-pages/pull/14

**Pull request merged**